src/Core/System/SalesChannel/Context/SalesChannelContextPersister.php

```diff
@@ -144,7 +144,11 @@ public function load(string $token, string $salesChannelId, ?string $customerId
         $now = new \DateTimeImmutable();
         if ($expiredTime < $now) {
             // context is expired
-            $payload = ['expired' => true];
+            if (!empty($customerId)) {
```

Comment:
```suggestion
            if ($customerId !== null) {
```

the same condition is used above. I do not like empty as check. we should be more explicit, what we want to check and need

Reply:
we are already protected by argument type
```
?string $customerId
```
But it doesn't hurt to be more specific, sure 👍